src/main/kotlin/com/exactpro/th2/check1/CollectorService.kt

```diff
@@ -97,7 +86,8 @@ class CollectorService(
 
         val chainID = request.getChainIdOrGenerate()
 
-        val task = CheckRuleTask(request.description, Instant.now(), SessionKey(sessionAlias, direction), request.timeout, maxEventBatchContentSize,
+        val task = CheckRuleTask(request.description, Instant.now(), SessionKey(sessionAlias, direction),
+            request.messageTimeout, request.timeout, maxEventBatchContentSize,
```

Comment:
Maybe group messageTimeout and timeout into an object and pass it as a parameter instead of two separate values?